Add messaging for Publisher; other RMQ updates

iainrb opened the pull request:
- Add RabbitMQ messaging for the WTSI::NPG::iRODS::Publisher class
- Modify the publish methods in WTSI::NPG::iRODS::Publisher to return a DataObject or Collection instead of a string
- Remove the list_path_details method from iRODS.pm; instead use other iRODS methods which can read cached results
- Need to explicitly call rmq_init and rmq_disconnect (no longer part of BUILD/DEMOLISH methods)
- Declare a separate RabbitMQ channel for each test
- Added tests for Publisher messaging and DataObject/Collection return values
- ReporterTest.pm improvements; skip tests for undefined input, sanity check on message queue status, sort AVUS for consistent order

Commit:
Add messaging for Publisher; change return values for publish methods; other RMQ updates

Commit:
Resolve conflicts introduced by merging PR#150

Commit:
Correct method names in Perldoc

keithj commented:
Class naming convention used in tests: class `Foo` has test class `FooTest`. `TestMQPublisher` and `TestMQiRODS` are not expected to contain unit tests, but to contain helper classes to be used in tests (rather like `TestCommunicator`).

Commit:
Refactored RabbitMQ tests; added checks on message body; no_rmq flag replaced by enable_rmq; test classes renamed

iainrb commented:
1. RabbitMQ tests enabled for Travis. Skipping tests now prints a diagnostic message (including in non-verbose mode).

2. Added a logwarn() message to make skip behaviour more clear. If a RabbitMQ message is undefined, subsequent tests on the message content are skipped to improve readability. This is distinct from skipping *all* RabbitMQ tests based on environment variable configuration (as in item 1).

3. Refactored tests. The _test_message private method now takes a HashRef argument containing an expected value for the message body, which it compares with the actual value. Specifically, it checks that the collection, data object name (if any) and AVUs (if any) are equivalent.

4. Replaced no_rmq flag with enable_rmq, which defaults to True. (Replaces the old default, in which no_rmq depended on whether rmq_config_path was defined.)

5. Test classes renamed as iRODSMQTest and PublisherMQTest.

6. Replaced with the simpler 'use Foo;'

7. Tabs replaced with spaces.
